### ISSUE TITLE
MangoHud freezes Vulkan applications with nVidia 550.78

### ISSUE THREAD

**Issue by VortexAcherontic** [flightlessmango/MangoHud#1314]
**Describe the bug**
As of today (yesterday everything was fine) all of the sudden MangoHUD freezes Vulkan based applications. Either when opening them up or closing them down.
I tested it with GZDoom using it's Vulkan backend. Which can be run just fine with MangoHUD enabled until you close it. Other apps like games using DXVK, Godot 4 will freeze right at the start as soon as the game window pops open.

Edit: Found out this happens only on Wayland. Running Gnome 46 in X11 does not cause the freezes. Switching to Gnome 46 Wayland does.

`Mai 01 09:39:03 kleinerpoppel org.zdoom.GZDoom.desktop[26609]: [*** LOG ERROR #0001 ***] [2024-05-01 09:39:03] [MANGOHUD] string pointer is null [../src/shared_x11.cpp(39)]
`

Disabling MangoHUD restored all Vulkan apps.

What changed is that my distribution updated the nvidia driver package to 550.78

```
Apr 30 18:31:49 kleinerpoppel transactional-update[6639]: Computing distribution upgrade...
Apr 30 18:31:50 kleinerpoppel transactional-update[6639]: The following 9 packages are going to be upgraded:
Apr 30 18:31:50 kleinerpoppel transactional-update[6639]:   MicroOS-release MicroOS-release-appliance nvidia-compute-G06 nvidia-compute-utils-G06 nvidia-driver-G06-kmp-default nvidia-drivers-G06 nvidia-gl-G06 nvidia-utils-G06 nvidia-video-G06
Apr 30 18:31:50 kleinerpoppel transactional-update[6639]: The following product is going to be upgraded:
Apr 30 18:31:50 kleinerpoppel transactional-update[6639]: openSUSE MicroOS
Apr 30 18:31:50 kleinerpoppel transactional-update[6639]:   20240428-0 -> 20240429-0
Apr 30 18:31:50 kleinerpoppel transactional-update[6639]: 9 packages to upgrade.
Apr 30 18:31:50 kleinerpoppel transactional-update[6639]: Overall download size: 235.0 MiB. Already cached: 0 B. After the operation, additional 411.7 KiB will be used.
...
Apr 30 18:31:51 kleinerpoppel transactional-update[6639]: Retrieving: nvidia-driver-G06-kmp-default-550.78_k6.8.7_1-22.1.x86_64 (repo-non-free) (1/9),  40.0 MiB
Apr 30 18:31:52 kleinerpoppel transactional-update[6639]: Retrieving: nvidia-driver-G06-kmp-default-550.78_k6.8.7_1-22.1.x86_64.rpm [.........done (36.3 MiB/s)]
```

**List relevant hardware/software information**
- Linux Distribution: openSUSE Aeon
- MangoHud version: 0.7.1
- GPU: nVidia RTX 3080 (Driver 550.78)

**To Reproduce**
1. Update nvidia driver to 550.78
2. Run Gnome 46 in Wayland mode
3. Enable MangoHUD
4. Open up any Vulkan application
5. Some freeze right away some later upon exiting

**Expected behavior**
Vulkan apps not to freeze

**Additional context**
It might be worth mentioning I run everything from flatpak. MangoHUD, GZDoom etc.
Verified all flatpak runtimes are up-to-date as well.

**Comment by flightlessmango** [comment]
I've attempted to fix this issue here a0a31b4

**Comment by BeauTaapken** [comment]
I'll give the git version a try, I'll let you know if this fixes my issue

**Comment by BeauTaapken** [comment]
> I've attempted to fix this issue here [a0a31b4](https://github.com/flightlessmango/MangoHud/commit/a0a31b4f64384c980b04216c806726da5b45f5cc)

I just tried with a build from the git version, but I still have the problem of my game just completely freezing before even rendering anything.

This is what my screen looks like after trying to start any game with mangohud enabled
![image](https://github.com/flightlessmango/MangoHud/assets/25839757/5290a9c0-1d63-4b27-b2f7-97e417e6a81c)

I also have this issue with Baldur's gate 2 enhanced edition, without enabling the alternative rendering mode, which should mean it's using opengl and not vulkan (unless proton does something i'm not aware of). Mangohud  does seem to at least load data before the game freezes, but I don't know if that has anything to do with the  issue. 
![image](https://github.com/flightlessmango/MangoHud/assets/25839757/a5a592ca-af8f-492c-8bdb-97f55cb6e0c1)


Please let me know if you need anything else to figure out the issue

**Comment by BeauTaapken** [comment]
I switched over to the nvidia 555 beta drivers. Something did change in the behaviour when mangohud is enabled, and it's slightly less annoying. The game just doesn't open now. Steam says the game is running, but the game never really starts.

**Comment by VortexAcherontic** [comment]
Can confrim as ell. With 555.42.02 it no longer freezes applications anymore. Maybe it was just a driver bug?

**Comment by Mart-Bogdan** [comment]
Doest this also happen under Wayland + KDE?  As, unlike X11, wayland has different implementation in different DE.

**Comment by VortexAcherontic** [comment]
Hi, there. Just wanted to let you know that this didn't happen again for a long time. Probably it was either a gnome bug or driver bug which was solved.

I'm closing here as this seems no longer relevant to me.

**Comment by flightlessmango** [comment]
I believe this was fixed in bda2af47b032dcec810203d34a486562176afb0e
